src/aestream.cpp

```diff
@@ -142,6 +163,12 @@ int main(int argc, char *argv[]) {
     input_generator = file_event_generator(input_filename, input_ignore_time);
   }
 
+  //
+  // Processing - Undistortion
+  //
+  Generator<AEDAT::PolarityEvent> processed_generator;
+  processed_generator = undistortion_event_generator(input_generator, undistortion_filename, width, height);
```

Comment:
Would it make sense to skip this step if the undistortion filter isn't used? I imagine it could generate some overhead if we pass the events through a second coroutine that doesn't do anything but return the events. 

What about something like
```
if (undistortion_filename == "") {
  processed_generator = input_generator;
} else { ... }
```

Reply:
That totally makes sense,. However,  I tried it and I got the following error during compilation:

```
[1/4] Building CXX object src/CMakeFiles/aestream_lib.dir/aestream.cpp.o
FAILED: src/CMakeFiles/aestream_lib.dir/aestream.cpp.o 
/usr/bin/c++ -DBOOST_ALL_NO_LIB -DBOOST_CHRONO_DYN_LINK -DBOOST_FILESYSTEM_DYN_LINK -DBOOST_TIMER_DYN_LINK -DWITH_CAER -DWITH_METAVISION -Daestream_lib_EXPORTS -I/home/juan/aestream/include -I/home/juan/aestream/src -I/home/juan/aestream/build/src/fb_headers -I/home/juan/aestream/build/_deps/flatbuffers-src/include -isystem /usr/include/opencv4 -fPIC   -fcoroutines -std=gnu++2a -MD -MT src/CMakeFiles/aestream_lib.dir/aestream.cpp.o -MF src/CMakeFiles/aestream_lib.dir/aestream.cpp.o.d -o src/CMakeFiles/aestream_lib.dir/aestream.cpp.o -c /home/juan/aestream/src/aestream.cpp
/home/juan/aestream/src/aestream.cpp: In function ‘int main(int, char**)’:
/home/juan/aestream/src/aestream.cpp:170:25: error: use of deleted function ‘Generator<T>& Generator<T>::operator=(const Generator<T>&) [with T = AEDAT::PolarityEvent]’
  170 |   processed_generator = input_generator;
      |                         ^~~~~~~~~~~~~~~
In file included from /home/juan/aestream/src/aedat4.hpp:23,
                 from /home/juan/aestream/src/aestream.cpp:11:
/home/juan/aestream/src/generator.hpp:39:14: note: declared here
   39 |   Generator &operator=(const Generator &) = delete;
      |              ^~~~~~~~
[2/4] Building CXX object src/CMakeFiles/aestream.dir/aestream.cpp.o
FAILED: src/CMakeFiles/aestream.dir/aestream.cpp.o 
/usr/bin/c++ -DBOOST_ALL_NO_LIB -DBOOST_CHRONO_DYN_LINK -DBOOST_FILESYSTEM_DYN_LINK -DBOOST_TIMER_DYN_LINK -DWITH_CAER -DWITH_METAVISION -I/home/juan/aestream/include -I/home/juan/aestream/src -I/home/juan/aestream/build/src/fb_headers -I/home/juan/aestream/build/_deps/flatbuffers-src/include -isystem /usr/include/opencv4 -fcoroutines -std=gnu++2a -MD -MT src/CMakeFiles/aestream.dir/aestream.cpp.o -MF src/CMakeFiles/aestream.dir/aestream.cpp.o.d -o src/CMakeFiles/aestream.dir/aestream.cpp.o -c /home/juan/aestream/src/aestream.cpp
/home/juan/aestream/src/aestream.cpp: In function ‘int main(int, char**)’:
/home/juan/aestream/src/aestream.cpp:170:25: error: use of deleted function ‘Generator<T>& Generator<T>::operator=(const Generator<T>&) [with T = AEDAT::PolarityEvent]’
  170 |   processed_generator = input_generator;
      |                         ^~~~~~~~~~~~~~~
In file included from /home/juan/aestream/src/aedat4.hpp:23,
                 from /home/juan/aestream/src/aestream.cpp:11:
/home/juan/aestream/src/generator.hpp:39:14: note: declared here
   39 |   Generator &operator=(const Generator &) = delete;
```

I'll check how to do it properly whenever I have time :)